### PR TITLE
Switch to the maven-publish plugin

### LINKED ISSUE #287: Replace gradle-mvn-push
The `maven` plugin on which our release script is built is scheduled to be removed in Gradle 7.0.  Time to move to the (gag) `maven-publish` plugin instead, somehow, while still remembering to ship code AND sources AND javadoc artifacts, all properly signed.

### PR DESCRIPTION
**Pull request by benjamin-bader**
@jaredsburrows not sure whether you've used `maven-publish` much, but if so I'd appreciate your once-over.  It seems to work (artifacts show up in Sonatype with signature files, etc), but who knows what might be missing/wrong here that could bit later.

The original `maven` plugin is officially deprecated now, so I'd prefer we cut over sooner rather than later.

Fixes #287

### CODE REVIEW

File: gradle/dependencies.gradle [hunk 1]
```diff
@@ -1,9 +1,27 @@
+/*
+ * Copyright 2019 Benjamin Bader
+ * Copyright 2017 Jared Burrows
```
Comment (jaredsburrows) [inline gradle/dependencies.gradle]: Thanks!

File: gradle/maven-publish.gradle [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    ]
+    deploymentTasks.any { project.gradle.taskGraph.hasTask(it) }
+}
+
+task sourceJar(type: Jar) {
+    archiveClassifier.set("sources")
+    from sourceSets.main.allSource
+}
+
+task javadocJar(type: Jar, dependsOn: dokka) {
```
Comment (jaredsburrows) [inline gradle/maven-publish.gradle]: Do you need one for dokka? 


```
task docsJar(type: Jar, dependsOn: dokka) {
  group = 'Publications'
  description = 'Create jar of documentation.'
  archiveClassifier = 'docs'
  from dokka.outputDirectory
}
See: https://github.com/jaredsburrows/gradle-license-plugin/blob/master/gradle/publish.gradle#L35
```

Comment (benjamin-bader) [reply gradle/maven-publish.gradle]: I don't think so?  This change uses Dokka for all documentation, replacing (or just ignoring) the Javadoc and Groovydoc jars we were generating before.  At the bottom of this task you can see that it's set up to contain only dokka output.  I might be missing something though, what do you think?

It's how I did it in Thrifty, so I just copypasta'd: https://github.com/microsoft/thrifty/blob/master/gradle/gradle-mvn-push.gradle#L108.

File: gradle/dependencies.gradle (hunk 1 above)
Comment (benjamin-bader) [reply gradle/dependencies.gradle]: Hey, you did all the hard work :)

File: gradle/maven-publish.gradle (hunk 2 above)
Comment (jaredsburrows) [reply gradle/maven-publish.gradle]: Oh sorry, I read "javadoc" and thought of Java.

**Pull request merged**